tests/sentry/api/endpoints/test_monitor_details.py

```diff
@@ -1,135 +1,197 @@
 
 from sentry.models import Monitor, MonitorStatus, MonitorType, ScheduledDeletion, ScheduleType
 from sentry.testutils import APITestCase
+from sentry.testutils.silo import region_silo_test
 
 
+@region_silo_test(stable=True)
 class MonitorDetailsTest(APITestCase):
-    def test_simple(self):
-        user = self.create_user()
-        org = self.create_organization(owner=user)
-        team = self.create_team(organization=org, members=[user])
-        project = self.create_project(teams=[team])
-
-        monitor = Monitor.objects.create(
-            organization_id=org.id,
-            project_id=project.id,
+    def setUp(self):
+        self.user = self.create_user()
+        self.org = self.create_organization(owner=self.user)
+        self.team = self.create_team(organization=self.org, members=[self.user])
+        self.project = self.create_project(teams=[self.team])
+
+        self.monitor = Monitor.objects.create(
+            organization_id=self.org.id,
+            project_id=self.project.id,
             next_checkin=timezone.now() - timedelta(minutes=1),
             type=MonitorType.CRON_JOB,
             config={"schedule": "* * * * *"},
         )
 
-        self.login_as(user=user)
+        self.paths = (
+            f"/api/0/monitors/{self.monitor.guid}/",
+            f"/api/0/monitors/{self.org.slug}/{self.monitor.guid}/",
+        )
+
+    def test_simple(self):
+        self.login_as(user=self.user)
+
         with self.feature({"organizations:monitors": True}):
-            resp = self.client.get(f"/api/0/monitors/{monitor.guid}/")
+            for path in self.paths:
+                resp = self.client.get(path)
+
+                assert resp.status_code == 200, resp.content
+                assert resp.data["id"] == str(self.monitor.guid)
 
-        assert resp.status_code == 200, resp.content
-        assert resp.data["id"] == str(monitor.guid)
+    def test_mismatched_org_slugs(self):
+        path = f"/api/0/monitors/asdf/{self.monitor.guid}/"
+        self.login_as(user=self.user)
 
+        with self.feature({"organizations:monitors": True}):
+            resp = self.client.get(path)
 
+            assert resp.status_code == 400
+
+
+@region_silo_test(stable=True)
 class UpdateMonitorTest(APITestCase):
     def setUp(self):
         self.user = self.create_user()
         self.org = self.create_organization(owner=self.user)
         self.team = self.create_team(organization=self.org, members=[self.user])
         self.project = self.create_project(teams=[self.team])
+        self.num_paths = 2
 
-        self.monitor = Monitor.objects.create(
+        self.login_as(user=self.user)
+
+    def _get_urls(self):
+        return ("sentry-api-0-monitor-details", "sentry-api-0-monitor-details-with-org")
+
+    def _get_path(self, i, monitor):
+        urls = self._get_urls()
+        if i:
+            return reverse(urls[i], args=[self.org.slug, monitor.guid])
+        return reverse(urls[i], args=[monitor.guid])
```

Comment:
This strikes me as awkward because the bodies of `_get_urls` and `_get_path` are tightly coupled: you have to know that the URL at index 0 takes an org slug and the one at index 1 doesn't. But I can see how it isn't that simple to fix, because there's variation in the way you set up the monitor.

One way to combine the two that I like would be
```python
    def _get_path_functions(self):
        return (
            lambda monitor: reverse("sentry-api-0-monitor-details", args=[monitor.guid]),
            lambda monitor: reverse(
                "sentry-api-0-monitor-details-with-org", args=[self.org.slug, monitor.guid]
            ),
        )
```
It's a bit more abstract, but note how it puts the org slug right next to the "-with-org" mapping.

To use it, you would change a block of code that looks like
```python
            for i in range(self.num_paths):
                monitor = self._create_monitor()
                path = self._get_path(i, monitor)
```
to
```python
            for path_func in self._get_path_functions():
                monitor = self._create_monitor()
                path = path_func(monitor)
```

Reply:
this is much cleaner 👍